projects/driver_controls/test/test_can_output.c

```diff
@@ -0,0 +1,299 @@
+// Test that the CAN output routines are properly generated once the correct events occur
+
+#include "event_arbiter.h"
+#include "event_queue.h"
+#include "input_event.h"
+#include "log.h"
+#include "status.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+#include "direction_fsm.h"
+#include "hazard_light_fsm.h"
+#include "horn_fsm.h"
+#include "mechanical_brake_fsm.h"
+#include "pedal_fsm.h"
+#include "power_fsm.h"
+#include "push_to_talk_fsm.h"
+#include "turn_signal_fsm.h"
+
+#include "can_output.h"
+
+typedef struct FSMGroup {
+  FSM power;
+  FSM pedal;
+  FSM direction;
+  FSM turn_signal;
+  FSM hazard_light;
+  FSM mechanical_brake;
+  FSM horn;
+  FSM push_to_talk;
+} FSMGroup;
+
+static FSMGroup s_fsm_group;
+static bool s_powered = false;
+static bool s_mech_brake = false;
+
+static EventArbiterOutputData s_can_output;
+
+static void prv_output(EventArbiterOutputData data) {
+  s_can_output.id = data.id;
+  s_can_output.state = data.state;
+  s_can_output.data = data.data;
+
+  LOG_DEBUG("[ .id = %x, .state = %x, .data = %x ]\n",
+          s_can_output.id,
+          s_can_output.state,
+          s_can_output.data);
+}
+
+static void prv_toggle_power(bool new_state) {
+  Event e = { .data = 0 };
+  if (new_state != s_powered) {
+    e.id = INPUT_EVENT_POWER;
+    event_arbiter_process_event(&e);
+    s_powered = new_state;
+  }
+}
+
+static void prv_toggle_mech_brake(bool new_state) {
+  Event e = { .data = 0 };
+
+  e.id = (new_state == true) ? INPUT_EVENT_MECHANICAL_BRAKE_PRESSED
```

Comment:
The `== true` is redundant since it is already a bool.

Reply:
Done